File: Union-of-2-Sorted-Arrays.py
class Solution:
    def findUnion(self, a, b):
        # code here 
        unionArr=[]
        n1=len(a)
        n2=len(b)
        i=0 
        j=0
        while (i<n1 and j<n2):
            if a[i]==b[j]:
                if (not unionArr or a[i] != unionArr[-1]):
                    unionArr.append(a[i]) 
                i+=1 
                j+=1
            elif a[i]<b[j]:
                if (not unionArr or a[i] != unionArr[-1]):
                    unionArr.append(a[i]) 
                i+=1 
            else:
                if (not unionArr or b[j] != unionArr[-1]):
                    unionArr.append(b[j]) 
                j+=1 
                
        while (j<n2):
            if (not unionArr or b[j] != unionArr[-1]):
                    unionArr.append(b[j]) 
            j+=1 
        while(i<n1):
            if (not unionArr or a[i] != unionArr[-1]):
                    unionArr.append(a[i]) 
            i+=1
        return unionArr
 

File: Union-of-2-Sorted-Arrays.js
/**
 * @param {number[]} a
 * @param {number[]} b
 * @returns {number[]}
 */
class Solution {
    findUnion(a, b) {
        // code here
        let unionArr=[]
        let n1=a.length
        let n2=b.length 
        let i=0
        let j=0 
        while (i<n1 && j<n2){
            if (a[i]===b[j]){
                if (a[i] !== unionArr[unionArr.length-1] || unionArr.length===0){
                    unionArr.push(a[i])
                }
                i++;
                j++;
            }
            else if (a[i]<b[j]){
                if (a[i] !== unionArr[unionArr.length-1] || unionArr.length===0){
                    unionArr.push(a[i])
                }
                i++;
            }
            else{
                if (b[j] !== unionArr[unionArr.length-1] || unionArr.length===0){
                    unionArr.push(b[j])
                }
                j++;
            }
        }
        while (i<n1){
            if (a[i] !== unionArr[unionArr.length-1] || unionArr.length===0){
                unionArr.push(a[i])
            }
            i++;
        }
        while (j<n2){
            if (b[j] !== unionArr[unionArr.length-1] || unionArr.length===0){
                unionArr.push(b[j])
            }
            j++;
        }
        return unionArr
    }
}

let a[] = [1, 2, 3, 4, 5], 
let b[] = [1, 2, 3, 6, 7]
console.log(new Solution().findUnion(a, b)) 
// Output: [1, 2, 3, 4, 5, 6, 7]
